### PR TITLE
fix: [PL-39440]: Added support to return delegate token value in list api

### PR DESCRIPTION
**Pull request by nishantsaini20**
## Describe your changes
The delegate token list API is expected to return `value` field as `null` when token details are not fetch by token name (i.e. `tokenName` is not passed in request).
Changes:
1. Added field description so that it is shown in API doc.
2. Added support in the API to return token value if user has edit delegate permission. Same permission is checked when fetching token details by token name.

Testing:
1. Verified the field description is coming in openapi.json
2. Verified list api works as following:
    - When user has edit delegate permission the api response has token value populated for each token
    - When user doesn't have edit permission but have view permission then the api response return token details but value of the token is not populated.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49393)

### CODE REVIEW

**Comment by nishantsaini20** [comment]
trigger uts

File: 120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java [hunk 1]
```diff
@@ -154,17 +154,21 @@ public DelegateTokenNgResource(
...
     if (StringUtils.isBlank(delegateTokenName)) {
       accessControlClient.checkForAccessOrThrow(ResourceScope.of(accountIdentifier, orgIdentifier, projectIdentifier),
           Resource.of(DELEGATE_RESOURCE_TYPE, null), DELEGATE_VIEW_PERMISSION);
+      hasEditAccess =
+          accessControlClient.hasAccess(ResourceScope.of(accountIdentifier, orgIdentifier, projectIdentifier),
+              Resource.of(DELEGATE_RESOURCE_TYPE, null), DELEGATE_EDIT_PERMISSION);
     } else {
       // When delegate token name is requested, token value will be included in response. Only roles with creating
       // a delegate can access.
       accessControlClient.checkForAccessOrThrow(ResourceScope.of(accountIdentifier, orgIdentifier, projectIdentifier),
```
Comment (xingchi-jin) [inline 120-ng-manager/src/main/java/io/harness/ng/core/delegate/resources/DelegateTokenNgResource.java:167]: how about we change API behavior to
when tokenName is provided -
1. if user has edit permission, provide token value
2. if user has only view permission, still return token information but without token value

**Comment by nishantsaini20** [comment]
trigger uts

**Comment by nishantsaini20** [comment]
trigger uts

**Comment by prashantsharma04** [comment]
trigger ut0

**Comment by nishantsaini20** [comment]
trigger uts

**Review by xingchi-jin — APPROVED** [review]

**Pull request merged**